### ISSUE TITLE
Error on photoshop 24.4.1

### ISSUE THREAD

**Issue by geangs** [zombieyang/sd-ppp#20]
I get this error when trying to run the Get Image From Photoshop node, do you know what could be causing this?

Error occurred when executing Get Image From Photoshop Layer:

sdppp PS side error:HTTP 500

File "--------\ComfyUI_windows_portable\ComfyUI\execution.py", line 152, in recursive_execute
output_data, output_ui = get_output_data(obj, input_data_all)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "--------\ComfyUI_windows_portable\ComfyUI\execution.py", line 82, in get_output_data
return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "--------\ComfyUI_windows_portable\ComfyUI\execution.py", line 75, in map_node_over_list
results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "--------\ComfyUI_windows_portable\ComfyUI\custom_nodes\sd-ppp\sdppp_python\comfy_nodes.py", line 90, in get_image
image_id, layer_opacity = call_async_func_in_server_thread(photoshopInstance.get_image(layer_id=id, bounds_id=bounds_id))
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "--------\ComfyUI_windows_portable\ComfyUI\custom_nodes\sd-ppp\sdppp_python\comfy_nodes.py", line 33, in call_async_func_in_server_thread
raise handle['error']
File "--------\ComfyUI_windows_portable\ComfyUI\custom_nodes\sd-ppp\sdppp_python\comfy_nodes.py", line 22, in do_call
handle['result'] = await coro
^^^^^^^^^^
File "--------\ComfyUI_windows_portable\ComfyUI\custom_nodes\sd-ppp\sdppp_python\photoshop_instance.py", line 59, in get_image
raise Exception('sdppp PS side error:' + result['error'])

**Comment by Jonseed** [comment]
Is it because the plugin is only compatible up to Photoshop 24.4.0?

**Comment by GeminiSquishGames** [comment]
The manifest says the minimum version is "minVersion": "25.4.0"... you could change that but there might be a requirement that 2024 has that 2023 lacks. I guess no one wants to support legacy versions of Photoshop? I basically have moved to Krita or Photopea, photoshop clone that works in the WebUI browser and ComfyUI.

**Comment by Jonseed** [comment]
I see minVersion as 24.4.0 in the manifest.
![Screenshot 2024-07-09 134214](https://github.com/zombieyang/sd-ppp/assets/40897189/05de5200-7c56-4697-8a43-11992d1ad4fa)

**Comment by Jonseed** [comment]
Or does that mean that is the minimum version (oldest) that is compatible. Newer versions should work?

**Comment by GeminiSquishGames** [comment]
> Or does that mean that is the minimum version (oldest) that is compatible. Newer versions should work?

I'm going by the release version's manifest. Not sure why there is a discrepancy...

It means that is the minimum/lowest version it will work with as per the developer's discretion. You can change it to bypass the automatic detection, say to 24.1.0, but it's usually set that way for good reason. #18 has that detail.

**Comment by zombieyang** [comment]
> Is it because the plugin is only compatible up to Photoshop 24.4.0?

yes. It will take some works to compat 24.4.0-. Still in progress yet

**Comment by zombieyang** [comment]
> I get this error when trying to run the Get Image From Photoshop node, do you know what could be causing this?
> 
> Error occurred when executing Get Image From Photoshop Layer:
> 
> sdppp PS side error:HTTP 500
> 
> File "--------\ComfyUI_windows_portable\ComfyUI\execution.py", line 152, in recursive_execute output_data, output_ui = get_output_data(obj, input_data_all) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "--------\ComfyUI_windows_portable\ComfyUI\execution.py", line 82, in get_output_data return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "--------\ComfyUI_windows_portable\ComfyUI\execution.py", line 75, in map_node_over_list results.append(getattr(obj, func)(**slice_dict(input_data_all, i))) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "--------\ComfyUI_windows_portable\ComfyUI\custom_nodes\sd-ppp\sdppp_python\comfy_nodes.py", line 90, in get_image image_id, layer_opacity = call_async_func_in_server_thread(photoshopInstance.get_image(layer_id=id, bounds_id=bounds_id)) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "--------\ComfyUI_windows_portable\ComfyUI\custom_nodes\sd-ppp\sdppp_python\comfy_nodes.py", line 33, in call_async_func_in_server_thread raise handle['error'] File "--------\ComfyUI_windows_portable\ComfyUI\custom_nodes\sd-ppp\sdppp_python\comfy_nodes.py", line 22, in do_call handle['result'] = await coro ^^^^^^^^^^ File "--------\ComfyUI_windows_portable\ComfyUI\custom_nodes\sd-ppp\sdppp_python\photoshop_instance.py", line 59, in get_image raise Exception('sdppp PS side error:' + result['error'])

But I think this issue might not related to version. comfyui should have been thrown more error except these message.

**Comment by zombieyang** [comment]
HTTP 500 bug was fixed in 
9b7c6c2
